### ISSUE TITLE
Custom metrics are not registered to the same registry

### ISSUE THREAD

**Issue by haakondr** [joao-fontenele/express-prometheus-middleware#22]
I attempt to setup custom metrics in combination with express-prometheus-middleware. My custom metric is counting, but is not served by the metrics endpoint, and from what I've understood I end up with two prometheus registers.

I made a minimal runnable example which can be seen at https://github.com/haakondr/prometheus-express-custom-metrics

My server is defined as follows:

```
const express = require('express');
const promMid = require('express-prometheus-middleware');
const client = require('prom-client');

const startServer = () => {
  const app = express();
  const port = 8080;

  const counter = new client.Counter({
    name: 'hellocounter',
    help: 'Counting hellos',
  });


  app.use(promMid({
    metricsPath: '/metrics',
    collectDefaultMetrics: true,
  }));


  app.get('/', (req, res) => {
    counter.inc(1);
    res.send(client.register.metrics());
  });

  return app.listen(port, () => {
    console.log(`Server started`);
  });

};


module.exports = {
  startServer: startServer,
};
```

With this setup, the default metrics from express-prometheus-middleware are served on /metrics, but not my hellocounter. GET / responds with my hellocounter, so it is being incremented.

When I debug in https://github.com/joao-fontenele/express-prometheus-middleware/blob/master/src/index.js#L89 I don't see my hellocounter, so I suspect that the problem is that both my custom metric and express-prometheus-middleware metrics are using global prometheus registers, but these are module scoped. When I have explicitly added prom-client as a dependency, I end up with two modules of prom-client:

```
npm ls | grep prom-client
│ ├─┬ prom-client@11.5.3
└─┬ prom-client@12.0.0
```

So, my question is as follows:

- what is the recommended approach for adding custom metrics?

One possible solution is to allow passing in prometheus registry as an option, and pass this along to Prometheus.collectDefaultMetrics in  https://github.com/joao-fontenele/express-prometheus-middleware/blob/master/src/index.js#L63. 

Eager to hear if you have a solution to this. If you think my suggestion is a viable solution, I can submit a PR.

**Comment by hlavacek** [comment]
Hi, I also got this problem some time ago - what helped me was to force the same dependency in package.json:
```
"prom-client": "^11.5.3",
```

and then run 
```
npm dedupe
```
after npm install ... I think also yarn would remove the duplicate dependencies automatically, then you have just single prom-client and it should work.

This is a workaround - I agree that a PR with passing registry would be better

**Comment by joao-fontenele** [comment]
hello @haakondr, 

It seems the latest commit of #23 fixes the issue, could you confirm? I've tried to clone and install the update, and it seems to work.

Just had to modify https://github.com/haakondr/prometheus-express-custom-metrics/blob/master/tests/test.js#L35 to

```

**Comment by enykeev** [comment]
@joao-fontenele I can confirm that #23 solves the problem. Thanks!

**Comment by joao-fontenele** [comment]
closed by #23
